front/lib/actions/mcp_internal_actions/output_schemas.ts

```diff
@@ -486,8 +487,51 @@ const NotificationTextContentSchema = z.object({
 
 type ImageProgressOutput = z.infer<typeof NotificationImageContentSchema>;
 
+// Schema for the resource of a notification where the tool is asking for tool approval.
+// This schema contains all the information that the MCP server runner
+// needs to emit an event for tool approval.
+const ToolApproveExecutionSchema = z.object({
+  type: z.literal("tool_approve_execution"),
+  configurationId: z.string(),
+  conversationId: z.string(),
+  messageId: z.string(),
+  actionId: z.string(),
+  inputs: z.record(z.unknown()),
+  stake: z.enum(MCP_TOOL_STAKE_LEVELS).optional(),
+  metadata: z.object({
+    mcpServerName: z.string(),
+    toolName: z.string(),
+    agentName: z.string(),
+  }),
+});
+
+type ToolApproveExecutionOutputType = z.infer<
+  typeof ToolApproveExecutionSchema
+>;
+
+const ToolApproveExecutionContentSchema = z.object({
+  type: z.literal("resource"),
```

Comment:
Hmmm not sure about this one why that resource indirection? You can rebase and look at run_agent.

Reply:
notifcation.output.data.resource is a bit crazy we can go directly for the content IMHO out of consistency